src/configs/battery.rs

```diff
@@ -24,17 +25,42 @@ impl<'a> Default for BatteryConfig<'a> {
             unknown_symbol: "",
             empty_symbol: "",
             format: "[$symbol$percentage]($style) ",
-            display: vec![BatteryDisplayConfig {
-                threshold: 10,
-                style: "red bold",
-            }],
+            display: vec![BatteryDisplayConfig::default()],
             disabled: false,
         }
     }
 }
 
-#[derive(Clone, ModuleConfig, Default, Serialize)]
+#[derive(Clone, Serialize)]
 pub struct BatteryDisplayConfig<'a> {
     pub threshold: i64,
     pub style: &'a str,
+    pub symbol: Option<&'a str>,
+}
+
+impl<'a> Default for BatteryDisplayConfig<'a> {
+    fn default() -> Self {
+        BatteryDisplayConfig {
+            threshold: 10,
+            style: "red bold",
+            symbol: None,
+        }
+    }
+}
+
+impl<'a> ModuleConfig<'a> for BatteryDisplayConfig<'a> {
+    fn from_config(config: &'a Value) -> Option<Self> {
+        let mut conf = BatteryDisplayConfig::default();
+
+        for (key, val) in config.as_table()?.iter() {
+            let s = key.to_string();
+            match &*s {
+                "threshold" => conf.threshold = val.as_integer()?,
+                "style" => conf.style = val.as_str()?,
+                "symbol" => conf.symbol = Some(val.as_str()?),
+                _ => (),
+            }
+        }
+        Some(conf)
+    }
```

Comment:
With the recent changes I don't think you need to implement this manually anymore.

Reply:
Thank you for teaching me.
I removed the above method.